gradle-conjure/src/test/groovy/com/palantir/gradle/conjure/ConjureServiceDependencyTest.groovy

```diff
@@ -256,11 +258,73 @@ class ConjureServiceDependencyTest extends IntegrationSpec {
         runTasksWithFailure(':api:generateConjureServiceDependencies')
     }
 
-    def readRecommendedProductDeps(File jarFile) {
+    def "no endpoint versions attribute if no min versions configured"() {
+        setup:
+        file('api/build.gradle') << '''
+        serviceDependencies {
+            serviceDependency {
+                productGroup = "com.palantir.conjure"
+                productName = "conjure"
+                minimumVersion = "1.2.0"
+                recommendedVersion = "1.2.0"
+                maximumVersion = "2.x.x"
+            }
+        }
+        '''.stripIndent()
+        when:
+        def result = runTasksSuccessfully(':api:api-jersey:Jar')
+
+        then:
+        Attributes attributes = getAttributes(file('api/api-jersey/build/libs/api-jersey-0.1.0.jar'))
+        attributes.containsKey(name(RecommendedProductDependencies.SLS_RECOMMENDED_PRODUCT_DEPS_KEY))
+        !attributes.containsKey(name(EndpointMinimumVersionsExtension.ENDPOINT_VERSIONS_MANIFEST_KEY))
+    }
+
+    def "endpoint information written if configured"() {
+        setup:
+        file('api/build.gradle') << '''
+        serviceDependencies {
+            serviceDependency {
+                productGroup = "com.palantir.conjure"
+                productName = "conjure"
+                minimumVersion = "1.2.0"
+                recommendedVersion = "1.2.0"
+                maximumVersion = "2.x.x"
+            }
+        }
+        endpointVersions {
```

Comment:
Similar to what Carter mentioned [here](https://github.com/palantir/gradle-conjure/pull/762#issuecomment-830177981), it feels a bit random to have this as its own top-level extension.

Should we move this inside the `serviceDependency` extension ([ref](https://github.com/palantir/gradle-conjure/blob/9752440246345e85add20605512ebb654bdd5abb/gradle-conjure-api/src/main/java/com/palantir/gradle/conjure/api/ConjureProductDependenciesExtension.java#L24))? E.g. something like:

```gradle
serviceDependencies {
    serviceDependency {
        // ...
    }
    endpointVersion {
       // ...
    }
}
```